provider/equinix/environ.go

```diff
@@ -0,0 +1,784 @@
+			Series:      args.InstanceConfig.Series,
+			Arches:      args.Tools.Arches(),
+			Constraints: args.Constraints,
+		},
+	)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if err := e.finishInstanceConfig(&args, spec); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	cloudCfg, err := cloudinit.New(args.InstanceConfig.Series)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	cloudCfg.AddScripts(
+		// This is a dummy script injected into packet images that
+		// confuses the init system detection logic used by juju.
+		"rm -f /sbin/initctl",
+	)
+
+	// Install additional dependencies that are present in ubuntu images
+	// but not in the versions built by equinix.
+	//
+	// NOTE(achilleasa): this is a hack and is only meant to be used
+	// temporarily; we must ensure that equinix mirrors the official
+	// ubuntu cloud images.
+	if _, err := series.UbuntuSeriesVersion(args.InstanceConfig.Series); err == nil {
+		cloudCfg.AddScripts(
+			"apt-get update",
+			"DEBIAN_FRONTEND=noninteractive apt-get --option=Dpkg::Options::=--force-confdef --option=Dpkg::Options::=--force-confold --option=Dpkg::Options::=--force-unsafe-io --assume-yes --quiet install dmidecode snapd lxd",
+		)
+	}
+
+	// NOTE(achilleasa): The following script applies a set of equinix-specific
+	// networking fixes:
+	//
+	// 1) The equinix provisioner creates a /8 route for private IP addresses.
+	// As a result, lxd is unable to find a suitable subnet to use when
+	// juju runs "lxd init --auto" and effectively prevents workloads from
+	// being deployed to equinix machines.
+	//
+	// The following fixup script queries the equinix metadata service and
+	// replaces these problematic routes with the correct ones based on the
+	// reserved block(s) that provide addresses to the machine.
+	//
+	// Another oddity inherent in equinix's network setup is that a sub-block
+	// is carved out of the reserved block and gets assigned to the machine.
+	// For instance, if the parent block is a /26, the machine gets a /31
+	// contained within the parent block. The route fixup script takes this
+	// into account and adds the right route based on the metadata details
+	// so we can route traffic to any other machine in the parent block.
+	//
+	// 2) The equinix provider requires the use of FAN for allowing
+	// container to container communication across nodes. Due to the way
+	// that the networking is configured, we need to install iptable rules
+	// to masquerade any non-FAN traffic so that containers get internet
+	// connectivity.  As Juju sets up FAN bridges via a worker, we run a
+	// script that waits for the bridge to appear and then makes the
+	// required iptable changes.
+	//
+	// The script is run once during provisioning and a cronjob is set up
+	// to ensure that it runs after each reboot.
+	cloudCfg.AddScripts(
+		`cat << 'EOF' >> /root/juju-fixups.sh
+		#!/bin/bash
+
+		curl -vs http://metadata.packet.net/metadata 2>/dev/null |
```

Comment:
Is this versioned on the packet side? If they change this, nobody can create an instance and we'll get spurious error messages during provisioning.

Reply:
yes it is